web/src/settings_org.js

```diff
@@ -686,6 +747,9 @@ export function discard_property_element_changes(elem, for_realm_default_setting
         case "realm_move_messages_between_streams_limit_seconds":
             set_msg_move_limit_setting(property_name);
             break;
+        case "realm_jitsi_server_url":
+            set_video_chat_provider_dropdown();
```

Comment:
Can't we call `set_jitsi_server_url_dropdown` here?

Reply:
Considering that the Jitsi URL setting is dependent on the video chat provider, to prevent bugs when resetting the Jitsi URL setting to its initial state upon clicking "Discard," we need to first reset the video chat provider dropdown. Using `set_jitsi_server_url_dropdown` or skipping the case would work just fine, given that `set_video_chat_provider_dropdown` will be called ahead in the code path for the video chat provider setting, thereby resetting both settings again.
```
case "realm_jitsi_server_url":
    set_jitsi_server_url_dropdown();
    break;
```
```
// skipping
case "realm_jitsi_server_url":
    break;
```

I was thinking, using `set_video_chat_provider_dropdown` here would make it self-sufficient. The alternative mentioned above would also work just fine. What do you think?

Reply:
Reading the code now, I am not sure why the code for calling `set_video_chat_provider_dropdown();` is in `update_dependent_subsettings`. I think `update_dependent_subsettings` is to update the elements which are not directly for the changed setting but to update the elements dependent on changed setting, which does not seem to be the case here.

I think the correct code here should be that the `case` block for `realm_video_chat_provider` should be in discard_property_element_changes` itself and then after that call you can add `case` block for `realm_jitsi_server_url` which would only call `set_jitsi_server_url_dropdown`.

Reply:
I'm not sure if this change should be a separate commit. For now, I've included the change in the same frontend commit.

Reply:
Would be good to have this in a separate commit. This commit already has many changes in `settings_org.js`.

Reply:
@sahil839 done.